### DISCUSSION TITLE
How to use the SaveSourceScreenshot call?

### DISCUSSION THREAD

**Discussion by CaptSzat** [Palakis/obs-websocket#1073] [Q&A]
I am trying to use the SaveSourceScreenshot but it keeps giving me a 702 error. 

Could anyone point me in the right direction? I know I am connecting to the websocket fine and I ahve been able to get other calls to work but I seem to be doing something wrong here. Here is the documentation https://github.com/obsproject/obs-websocket/blob/master/docs/generated/protocol.md#savesourcescreenshot . This is JS.

Heres my code,
```
let dataRequest = {
  sourceName: "Testing",
  imageFormat: 'png', 
  imageFilePath: "C:\test\screenshot.png"
}
....
await obs.call("SaveSourceScreenshot", dataRequest)
```

**Comment by northamerican** [comment]
seems `obs-websocket` fails to write to the file system. instead, get the raw data and write to a file yourself. here's what worked for me:

```js
import { OBSWebSocket } from "obs-websocket-js";
import fs from "fs";

const response = await obs.call("GetSourceScreenshot", {
  sourceName: "Testing",
  imageFormat: "png",
});

const buffer = Buffer.from(
  response.imageData.split("base64,")[1], // remove data URI
  "base64"
);

fs.writeFileSync(`C:\test\screenshot.png`, buffer);
```
